### PR TITLE
OSSL_STORE: Windows: use system certificates storage as default store

### PR DESCRIPTION
**Pull request by yjh-styx**
CLA: trivial

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/x509/by_store.c [hunk 1]
```diff
@@ -116,7 +116,11 @@ static int by_store_ctrl_ex(X509_LOOKUP *ctx, int cmd, const char *argp,
             argp = ossl_safe_getenv(X509_get_default_cert_dir_env());
 
         if (argp == NULL)
+#ifndef _WIN32
             argp = X509_get_default_cert_dir();
+#else
+            argp = "org.openssl.winstore://";
+#endif
```
Comment (tmshort) [inline crypto/x509/by_store.c:123]: Wondering why this isn't within `X509_get_default_cert_dir()`?

Comment (mattcaswell) [reply crypto/x509/by_store.c:123]: From the man page "The X509_get_default_cert_dir() function returns a default delimeter-separated
list of paths to a directories containing trusted CA certificates named in the hashed format" - so having it return a store URI would be kind of weird.

**Comment by t8m** [comment]
We did not make it a default store in 3.2 as there are some considerations around it. The problem is that there might be additional attributes stored along the cert itself in the Windows store that the current implementation ignores. This can have potential security implications. @hlandau @levitte might want to add more information.

File: crypto/x509/by_store.c (hunk 1 above)
Comment (levitte) [reply crypto/x509/by_store.c:123]: A `X509_get_default_cert_store()` would be more logical.

**Comment by levitte** [comment]
Why was this closed?

**Comment by yjh-styx** [comment]
Sorry - I'm can't sync repository (for create another pull request), and in comment was say's ~ "This variant is unacceptable..." and I think it not needed now and sync fork.
Do I recreate it?

**Pull request closed without merging**